contrib/win32/win32compat/ssh-agent/keyagent-request.c

```diff
@@ -113,6 +117,86 @@ convert_blob(struct agent_connection* con, const char *blob, DWORD blen, char **
 	return success? 0: -1;
 }
 
+/*
+ * in current_user sub tree under key_name key
+ * remove all sub keys with value name value_name_to_remove
+ * and value data value_data_to_remove
+ */
+static int
+remove_matching_subkeys_from_registry(HKEY user_root, wchar_t* key_name, wchar_t* value_name_to_remove, char *value_data_to_remove) {
+	int index = 0, success = 0;
+	DWORD data_len;
+	HKEY root = 0, sub = 0;
+	char *data = NULL;
+	wchar_t sub_name[MAX_KEY_LENGTH];
+	DWORD sub_name_len = MAX_KEY_LENGTH;
+
+	if (RegOpenKeyExW(user_root, key_name, 0, DELETE | KEY_ENUMERATE_SUB_KEYS | KEY_WOW64_64KEY, &root) != 0) {
+		success = 1;
```

Comment:
Since the removal didn't happen, I think any type of error should be propagated here. Let me know if that makes sense.
I've made a change to reflect that.